src/repository/scriptRepository.js

```diff
@@ -0,0 +1,83 @@
+const path = require('path');
+const Promise = require('bluebird');
+const fs = Promise.promisifyAll(require('node:fs'));
+
+const db = require('./database.js');
+const { Character, parseTitle, parseLines, parseCharacters } = require('../playData.js');
+
+function getCharacters(title, actorId = -1) {
+  const values = [actorId, title];
+  const sql = `
+      SELECT c.id as id, c.name as name, line_count, speaks_count
+      FROM characters c
+               JOIN scripts s ON script_id = s.id
+      WHERE ($1 = -1 or actor_id = $1)
+        AND s.title = $2`;
```

Comment:
New sql query can now filter the characters for a particular actor instead of having to do it in js. 
We use a sentinel value of `-1` for the actor id to indicate that we don't want to filter on actors and just want all characters.